[UPDATE] 닉네임 조회, 프로필 URL 조회, 작성한 게시글 수 조회, 작성한 프로젝트 수 조회, 설정 변경버튼

codero225 opened the pull request:
## 작업 이유
기능 구현

1. 유저 기본 정보(닉네임, 프로필 url) 조회
2. 유저 작성 게시글 수 조회
3. 유저 작성 프로젝트 수 조회
4. 유저가 소속된 그룹 조회
5. 본인 확인 후 설정변경 가능여부 확인

## 작업 사항
![image](https://github.com/80000Coding/80000Coding-Backend/assets/137995283/ddb9ecbd-1823-4ce6-9119-d2a511856e2e)

## 생각해볼 사항
access token의 required가 false가 될 수 있으면서 NPE 에러가 생김.

## 이슈 연결
close #45

codero225 committed:
Added Organization Domain and modified User Entity
To get organizations of the user

codero225 committed:
temp

codero225 committed:
test

codero225 committed:
Fixed minor issues
deleted and modified refresh signup token related files
add objectMapper in SecurityFilterConfig for correct initialization of JwtExceptionFilter
removed @Content in JwtExceptionFilter
modified comparison of role with name to code
add oauth info in application.yml

codero225 committed:
Fixed minor issues
deleted refresh signup token related methods
add objectMapper in SecurityFilterConfig for correct initialization of JwtExceptionFilter
removed content annotation in JwtExceptionFilter
modified comparison of role with name to code
add oauth info in application.yml

codero225 committed:
Merge branch 'feat/profile-info' into feat/dev-fix-with-profile-basic

codero225 committed:
Modified settings for profile servcie

codero225 committed:
Merge branch 'dev' into feat/dev-fix-with-profile-basic

codero225 committed:
[UPDATE] profile info 조회
setting 부분의 service인 UserProfileService는 ProfileServcie로 통합
NPE가 생기는 AccessTokenInfoResolver 주석처

codero225 committed:
[UPDATE] AuthController에서 response에 body가 없어지는 문제 해결

codero225 commented on src/main/java/io/oopy/coding/api/user/controller/UserAPI.java:
```diff
@@ -104,53 +104,4 @@ public ResponseEntity<?> authenticationTest(@AuthenticationPrincipal CustomUserD
 
         return ResponseEntity.ok(user);
     }
-
```
ProfileController에 합침

codero225 commented on src/main/java/io/oopy/coding/common/util/jwt/JwtUtilImpl.java:
```diff
@@ -37,22 +37,18 @@ public class JwtUtilImpl implements JwtUtil {
     private final Duration accessTokenExpirationTime;
     private final Duration refreshTokenExpirationTime;
     private final Duration signupAccessTokenExpirationTime;
```
사용하지 않는 singupRefreshToken 내용 삭제

ChanHoHan commented on src/main/java/io/oopy/coding/api/profile/service/ProfileService.java:
```diff
@@ -0,0 +1,103 @@
+            settingFlag = false;
+
+        return Map.of(
+                "setting_flag", settingFlag,
```
API ResponseBody에 Object타입 key에 snake case로 들어가는 건가요? 
만약 맞다면 camel case로 통일하는게 좋을거 같습니다 ! !

codero225 replied:
넵 변경 하였습니다

ChanHoHan commented on src/main/java/io/oopy/coding/domain/user/entity/User.java:
```diff
@@ -35,6 +40,15 @@ public class User extends Auditable {
     private RoleType role;
 
+    @Formula("(SELECT COUNT(*) FROM CONTENT C WHERE C.USER_ID = id AND C.CONTENT_TYPE = 'post' AND C.COMPLETE = 1 AND C.DELETE_DT IS NULL)")
+    private int postCount;
```
직접 얘기 나눈 부분이지만, comment를 남기겠습니다.
- eager 전략이면 user를 조회할때마다 해당 추가쿼리가 나가는지 확인 필요
- user의 직접적인 속성은 아닌거 같아서, 해당 엔티티에 이 쿼리가 있는것이 맞는지 의문 (제 개인적인 의견입니다)

codero225 replied:
의견 감사합니다~ 
새로운 commit에 해당부분 고려하여 userStatisticService를 따로 구현하였습니다!
그런데 쿼리를 따로 날리려면 설정이 필요하니 찬호님이 QueryDSL 적용 후 해당 부분 변경하겠습니다

codero225 committed:
📝 changed style of key of responseBody from snake case -> camel case

codero225 committed:
📰 Add userStatisticService for counting posts and projects
removed @Formula from domain User and moved to UserStatisticService
ignore profile related URL from security config files
add getter method in Organization and UserOrganization domain

codero225 commented on src/main/java/io/oopy/coding/api/user/service/UserStatisticService.java:
```diff
@@ -0,0 +1,24 @@
+@Service
+@Transactional
+@RequiredArgsConstructor
+public class UserStatisticService {
```
추후 QueryDSL 적용 후 변경 예정
